bash/drive-utils.bash

```diff
@@ -0,0 +1,529 @@
+    declare job="${1}" input="${2}" folder_id="${3}" token="${4}"
+    declare slug inputname extension inputsize readable_size request_method url postdata uploadlink upload_body mime_type resume_args
+
+    slug="${input##*/}"
+    inputname="${slug%.*}"
+    extension="${slug##*.}"
+    inputsize="$(($(wc -c < "${input}")))" && content_length="${inputsize}"
+    readable_size="$(_bytes_to_human "${inputsize}")"
+
+    # Handle extension-less files
+    [[ ${inputname} = "${extension}" ]] && declare mime_type && {
+        mime_type="$(file --brief --mime-type "${input}" || mimetype --output-format %m "${input}")" 2> /dev/null || {
+            "${QUIET:-_print_center}" "justify" "Error: file or mimetype command not found." "=" && printf "\n"
+            exit 1
+        }
+    }
+
+    _print_center "justify" "${input##*/}" " | ${readable_size}" "="
+
+    # Set proper variables for overwriting files
+    [[ ${job} = update ]] && {
+        declare existing_file_check_json
+        # Check if file actually exists, and create if not.
+        existing_file_check_json="$(_check_existing_file "${slug}" "${folder_id}" "${token}")"
+        if [[ -n ${existing_file_check_json} ]]; then
+            if [[ -n ${SKIP_DUPLICATES} ]]; then
+                # Stop upload if already exists ( -d/--skip-duplicates )
+                _collect_file_info "${existing_file_check_json}" "${slug}" || return 1
+                _clear_line 1
+                "${QUIET:-_print_center}" "justify" "${slug}" " already exists." "=" && return 0
+            else
+                request_method="PATCH"
+                _file_id="$(_json_value id 1 1 <<< "${existing_file_check_json}")"
+                url="${API_URL}/upload/drive/${API_VERSION}/files/${_file_id}?uploadType=resumable&supportsAllDrives=true"
+                # JSON post data to specify the file name and folder under while the file to be updated
+                postdata="{\"mimeType\": \"${mime_type}\",\"name\": \"${slug}\",\"addParents\": [\"${folder_id}\"]}"
+                STRING="Updated"
+            fi
+        else
+            job="create"
+        fi
+    }
+
+    # Set proper variables for creating files
+    [[ ${job} = create ]] && {
+        url="${API_URL}/upload/drive/${API_VERSION}/files?uploadType=resumable&supportsAllDrives=true"
+        request_method="POST"
+        # JSON post data to specify the file name and folder under while the file to be created
+        postdata="{\"mimeType\": \"${mime_type}\",\"name\": \"${slug}\",\"parents\": [\"${folder_id}\"]}"
+        STRING="Uploaded"
+    }
+
+    __file="${HOME}/.google-drive-upload/${slug}__::__${folder_id}__::__${inputsize}"
+    # https://developers.google.com/drive/api/v3/manage-uploads
+    if [[ -r "${__file}" ]]; then
+        uploadlink="$(< "${__file}")"
+        http_code="$(curl --compressed -s -X PUT "${uploadlink}" --write-out %"{http_code}")" || :
+        case "${http_code_upload_file}" in
```

Comment:
Codacy found an issue: [http_code_upload_file is referenced but not assigned.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)